### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by mllg**
* Be reasonable with dependencies. E.g., we do not need `stringr` for `str_detect`, just use `grepl()` instead.

* Some examples are in DONTRUN. Can we put them in `if (requireNamespace(...))` blocks instead?

* ResamplingSpCVBuffer looks similar to LOO. If this is the case, the instance should be stored more efficiently.

* autoplot tests are not working for me or are waaay to slow for unit tests (I'm stuck in there with 100% CPU)

* `blockCV::spatialBlock()` seems to call print(). It is convention (maybe even a CRAN policy?) to use `message()` which you can be suppressed. This should be reported upstream.

* `blockCV` is terrible slow (and has a stupid long dependency chain). Is there an alternative?

* Suggested packages should be explicitly attached with `require_namespaces`.

### CODE REVIEW

File: R/TaskClassifST.R [hunk 1]
```diff
@@ -130,6 +122,8 @@ TaskClassifST = R6::R6Class("TaskClassifST",
   catf(str_indent("Properties:", self$properties))
   catf(str_indent("Coordinates", self$coordinate_names))
 
+  # TODO: is this copy-pasted from Task?
+  # We should try to find a better way.
```
Comment (pat-s) [inline R/TaskClassifST.R]: Yes. With the aim of also printing the coords.

Maybe we can add an argument to `mlr3::task_print()` accepting a list of "type:column", i.e. `list(Coordinates = coordinate_names)` for internal extension capabilities?

File: R/Task_classif_ecuador.R [hunk 2]
```diff
@@ -23,8 +23,9 @@
 NULL
 
 load_task_ecuador = function(id = "ecuador") {
-  b = as_data_backend(readRDS(system.file("extdata", "ecuador.rda",
-    package = "mlr3spatiotemporal")))
+  data = readRDS(system.file("extdata", "ecuador.rda", package = "mlr3spatiotemporal"))
+  rownames(data) = NULL # TODO: better re-save the data set without rownames
```
Comment (pat-s) [inline R/Task_classif_ecuador.R]: Ahh, that's why we probably had so many troubles dealing with the row_ids...

File: DESCRIPTION [hunk 3]
```diff
@@ -26,17 +26,12 @@ Imports:
...
-    lgr,
     sf,
     stringr,
     testthat,
     vdiffr
-VignetteBuilder:
-    knitr
 Remotes:
     be-marc/blockCV
-ByteCompile: true
```
Comment (pat-s) [inline DESCRIPTION]: Ah yes, ByteCompile now defaults to true, doesn't it?

**Comment by pat-s** [comment]
> Be responsible with dependencies. E.g., we do not need stringr for str_detect, just use grepl() instead.

✅ 

> Some examples are in DONTRUN. Can we put them in if (requireNamespace(...)) blocks instead?

The reasoning was a runtime > 5s (I think 6s or so) triggering a note in R CMD check. This would not make a difference then right?

> ResamplingSpCVBuffer looks similar to LOO. If this is the case, the instance should be stored more efficiently.

It is in the end, just with the addition of a spatial buffer between train and test. I'll take a look how mlr3 stores LOOCV.

> autoplot tests are not working for me or are waaay to slow for unit tests (I'm stuck in there with 100% CPU)

They are costly yes because some tests are creating gridded plots of multiple folds. I do not think that there is another way here? I would like to avoid not testing them. 
For me they work and on Travis everything is okay as well. They take between 100-200s. Did you wait so long?

> blockCV::spatialBlock() seems to call print(). It is convention (maybe even a CRAN policy?) to use message() which you can be suppressed. This should be reported upstream.

Yes, I think there are more such possible issues. `blockCV` is not yet on CRAN.

> Suggested packages should be explicitly attached with require_namespaces.

✅

File: DESCRIPTION (hunk 3 above)
Comment (mllg) [reply DESCRIPTION]: Yep

File: R/Task_classif_ecuador.R (hunk 2 above)
Comment (mllg) [reply R/Task_classif_ecuador.R]: Should also work with rownames, but it is more efficient with integer.

File: R/TaskClassifST.R (hunk 1 above)
Comment (mllg) [reply R/TaskClassifST.R]: Yep, we maybe need to formalize the printing stuff more and implement a two-step approach: one function which converts into such a named list, and a second one for printing.

**Comment by mllg** [comment]
> The reasoning was a runtime > 5s (I think 6s or so) triggering a note in R CMD check. This would not make a difference then right?

Nope, it wouldn't. Maybe it is possible to find smaller tasks then?



> It is in the end, just with the addition of a spatial buffer between train and test. I'll take a look how mlr3 stores LOOCV.

There is no such thing currently. But LOO is the same thing as a n-fold CV where n is the number of observations. A CV is stored as (nx2) data.table, and you could do the same for the LOOCV (columns row_id + fold in which the row is in test).



> They are costly yes because some tests are creating gridded plots of multiple folds. I do not think that there is another way here? I would like to avoid not testing them.
> For me they work and on Travis everything is okay as well. They take between 100-200s. Did you wait so long?

TBH, no. I thought something was not working.



> Yes, I think there are more such possible issues. `blockCV` is not yet on CRAN.

Are you involved into this package?

**Comment by pat-s** [comment]
> Nope, it wouldn't. Maybe it is possible to find smaller tasks then?

I am not eager finding new tasks because the examples run 1s too long :) 
Ofc running them would be convenient to check for errors but I also do not think that they are prone to break soon. 

> Are you involved into this package?

No. Its just related to the work I do in science.

**Comment by nk027** [comment]
> > The reasoning was a runtime > 5s (I think 6s or so) triggering a note in R CMD check. This would not make a difference then right?

> Nope, it wouldn't. Maybe it is possible to find smaller tasks then?

Have a look at wrapping the examples or parts in `\dontrun{...}` instead. You can still test these with the flag `--run-donttest`.

**Comment by pat-s** [comment]
> Have a look at wrapping the examples or parts in \dontrun{...} instead. You can still test these with the flag --run-donttest.

Thanks. Was not aware of that flag.

**Comment by pat-s** [comment]
I'll move the "efficient LOOCV storage" to a separate issue and merge this one for now.
Thanks!!

**Pull request merged**